metrics-influxdb/src/main/java/io/dropwizard/metrics5/influxdb/InfluxDbReporter.java

```diff
@@ -0,0 +1,517 @@
+        }
+    }
+
+    private void writeMeteredFieldsIfEnabled(Metered meter) {
+        writeFieldIfEnabled(COUNT,     meter.getCount());
+        writeFieldIfEnabled(SUM,       meter.getSum());
+        writeFieldIfEnabled(M1_RATE,   convertRate(meter.getOneMinuteRate()));
+        writeFieldIfEnabled(M5_RATE,   convertRate(meter.getFiveMinuteRate()));
+        writeFieldIfEnabled(M15_RATE,  convertRate(meter.getFifteenMinuteRate()));
+        writeFieldIfEnabled(MEAN_RATE, convertRate(meter.getMeanRate()));
+    }
+    
+    private void reportCounter(MetricName name, Counter counter, long timestamp) throws IOException {
+        writeMeasurement(name);
+        writeFieldIfEnabled(COUNT, counter.getCount());
+        if (hasValues()) {
+            writeTimestampMillis(timestamp);
+            reportLine();
+        }
+    }
+
+    private void reportGauge(MetricName name, Gauge<?> gauge, long timestamp) throws IOException {
+        writeMeasurement(name);
+        Object value = gauge.getValue();
+        if (value != null) {
+            writeField(VALUE);
+            if (value instanceof Number) {
+                Number number = (Number) value;
+                if (number instanceof Long || 
+                        number instanceof Integer || 
+                        number instanceof Short || 
+                        number instanceof Byte) {
+                    writeFieldValue(number.longValue());
+                } else {
+                    writeFieldValue(number.doubleValue());
+                }
+            } else if (value instanceof Boolean) {
+                writeFieldValue(((Boolean)value));
+            } else {
+                writeFieldValue(value.toString());
+            }
+        }
+        writeTimestampMillis(timestamp);
+        reportLine();
+    }    
+    
+    private void reset() {
+        str.delete(0, str.length());
+    }
+    
+    private void writeMeasurement(MetricName name) {
+        reset();
+        char[] nameChars = encodedNameCache.get(name);
+        if (nameChars != null) {
+            str.append(nameChars);
+        } else {
+            writeMeasurementNoCache(name);
+            nameChars = new char[str.length()];
+            str.getChars(0, str.length(), nameChars, 0);
+            encodedNameCache.put(name, nameChars);
+        }
+        
+        str.append(' ');
+        firstField = true;
+    }
+    
+    private void writeMeasurementNoCache(MetricName name) {
+        MetricName prefixedName = MetricName.join(prefix, name);
+        
+        appendName(prefixedName.getKey(), str);
+        Map<String, String> tags = prefixedName.getTags();
+        if (!tags.isEmpty()) {
+            // InfluxDB Performance and Setup Tips:
+            // Sort tags by key before sending them to the database. 
+            // The sort should match the results from the Go bytes.Compare function.
+            TreeMap<String, String> sortedTags = new TreeMap<>(tags);
```

Comment:
I think we could take advantage of the Stream API and use it for building the metric name in a declarative fashion:
```java
MetricName prefixedName = MetricName.join(prefix, name);
return prefixedName.getTags()
          .entrySet()
          .stream()
          .sorted(Comparator.comparing(Map.Entry::getKey))
          .reduce(appendName(prefixedName.getKey(), new StringBuilder()), (sb, e) -> {
                    sb.append(',');
                    appendName(e.getKey(), sb);
                    sb.append('=');
                    return appendName(e.getValue(), sb);
          }, (l, r) -> l)
          .toString();
```